metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -61,47 +62,60 @@ final class Embedded(
 }
 
 object Embedded {
-  def downloadSettings(
-      dependency: Dependency,
-      scalaVersion: String
-  ): Settings =
-    new coursiersmall.Settings()
-      .withTtl(Some(Duration.Inf))
-      .withDependencies(List(dependency))
-      .addRepositories(
-        List(
-          coursiersmall.Repository.SonatypeReleases,
-          coursiersmall.Repository.SonatypeSnapshots
+  lazy val repositories: mutable.Buffer[Repository] =
+    Repository.defaults().asScala ++
+      List(
+        Repository.central(),
+        Repository.ivy2Local(),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/releases/"
+        ),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/snapshots/"
         )
       )
-      .withForceVersions(
-        List(
-          new Dependency(
-            "org.scala-lang",
-            "scala-library",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-compiler",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-reflect",
-            scalaVersion
-          )
-        )
+
+  def dependency(dep: String): Dependency = {
+    val regex = """([^:]+):([^:]+):([^:]+)""".r
+
+    dep match {
+      case regex(org, name, version) =>
```

Comment:
what if the regex doesn't match?

Reply:
It's just a test, so I wouldn't bother validating this. You can optionally log the unexpected case when the regex doesn't match

Reply:
If regex doesn't match then the expect part of the test is malformed, it doesn't indicate that the tested feature is wrong.

Reply:
Ok, this is not a test. Doing regex without the actual need for them is bad. Let's revert.

Reply:
Let's keep the dependencies in one like `Dependency.of("foo", "bar", "baz")` and either keep the regex only in the test scope or drop them altogether